service/awsconfig/v8/resource/lifecycle/current.go

```diff
@@ -95,34 +101,212 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+
+				r.logger.LogCtx(ctx, "level", "debug", "message", "inspected node config for the guest cluster")
+			}
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "no node configs to inspect for the guest cluster")
+		}
 	}
 
 	return nil, nil
 }
+
+func (r *Resource) completeLifecycleHook(ctx context.Context, instanceID, workerASGName string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("completing lifecycle hook action for guest cluster instance '%s'", instanceID))
+
+	i := &autoscaling.CompleteLifecycleActionInput{
+		AutoScalingGroupName:  aws.String(workerASGName),
+		InstanceId:            aws.String(instanceID),
+		LifecycleActionResult: aws.String("CONTINUE"),
+		LifecycleHookName:     aws.String(key.NodeDrainerLifecycleHookName),
+	}
+
+	_, err := r.aws.AutoScaling.CompleteLifecycleAction(i)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("completed lifecycle hook action for guest cluster instance '%s'", instanceID))
+
+	return nil
+}
+
+func (r *Resource) createNodeConfig(ctx context.Context, customObject providerv1alpha1.AWSConfig, instanceID, privateDNS string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "creating node config for guest cluster node")
+
+	n := v1.NamespaceDefault
+	c := &v1alpha1.NodeConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				key.InstanceIDAnnotation: instanceID,
+			},
+			Labels: map[string]string{
+				key.ClusterIDLabel: key.ClusterID(customObject),
+			},
+			Name: privateDNS,
+		},
+		Spec: v1alpha1.NodeConfigSpec{
+			Guest: v1alpha1.NodeConfigSpecGuest{
+				Cluster: v1alpha1.NodeConfigSpecGuestCluster{
+					API: v1alpha1.NodeConfigSpecGuestClusterAPI{
+						Endpoint: key.ClusterAPIEndpoint(customObject),
+					},
+					ID: key.ClusterID(customObject),
+				},
+				Node: v1alpha1.NodeConfigSpecGuestNode{
+					Name: privateDNS,
+				},
+			},
+			VersionBundle: v1alpha1.NodeConfigSpecVersionBundle{
+				Version: "0.1.0",
+			},
+		},
+	}
+
+	_, err := r.g8sClient.CoreV1alpha1().NodeConfigs(n).Create(c)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "created node config for guest cluster node")
+
+	return nil
+}
+
+func (r *Resource) deleteNodeConfig(ctx context.Context, privateDNS string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "deleting node config for guest cluster node")
+
+	n := v1.NamespaceDefault
+	i := privateDNS
+	o := &metav1.DeleteOptions{}
+
+	err := r.g8sClient.CoreV1alpha1().NodeConfigs(n).Delete(i, o)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "deleted node config for guest cluster node")
+
+	return nil
+}
+
+func (r *Resource) privateDNSForInstance(instanceID string) (string, error) {
```

Comment:
Here we need some EC2 magic. AutoScaling does not know about certain instance details.